x-pack/plugins/canvas/public/components/sidebar_header/sidebar_header.tsx

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { PureComponent, Fragment, MouseEvent } from 'react';
+import PropTypes from 'prop-types';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiButtonIcon,
+  EuiContextMenu,
+  EuiToolTip,
+} from '@elastic/eui';
+// @ts-ignore
+import { flattenPanelTree } from '../../lib/flatten_panel_tree';
+// @ts-ignore
+import { Popover } from '../popover';
+import { CustomElementModal } from './custom_element_modal';
+
+const topBorderClassName = 'canvasContextMenu--topBorder';
+
+export interface Props {
+  title: string;
+  showLayerControls?: boolean;
+  cutNodes: () => void;
+  copyNodes: () => void;
+  pasteNodes: () => void;
+  cloneNodes: () => void;
+  deleteNodes: () => void;
+  bringToFront: () => void;
+  bringForward: () => void;
+  sendBackward: () => void;
+  sendToBack: () => void;
+  createCustomElement: () => void;
```

Comment:
All of these handlers take no arguments and return nothing?

Reply:
Yes, that's correct. They're all functions handlers returned by the element handler creators in `public/lib/element_handler_creators`. Everything they need is passed as props.

I added comments for each handler.